### DISCUSSION TITLE
How to render sidecar file dynamically by RAW and XMP file and ignore JPEG which alone with the RAW file in originals.

### DISCUSSION THREAD

**Discussion by StephanXu** [photoprism/photoprism#4376] [General]
All of my photos are token in 'RAW+JPEG' so every raw file in originals directory has a jpeg file with the same name.

My adjustment in Lightroom will be saved in XMP file along with the RAW and the JPEG file. The JPEG file won't be overwritten. But PhotoPrism always see my JPEG file as the final look while actually XMP has the latest adjustments.

In my opinion, RAW+XMP is just fine for generating sidecar file, and I really want it re-generate sidecar file for syncing the adjustment I made.

Will it be possible?

**Comment by graciousgrey** [comment]
JPEGs are currently not regenerated when related RAW or XMP files change.

We recommend manually updating existing JPEG sidecar files as needed or creating additional JPEGs, so you can choose between different versions. New files and other metadata changes are detected and reflected in the index as usual when your library is scanned.

https://docs.photoprism.app/getting-started/faq/#will-jpegs-be-updated-when-the-related-raw-or-xmp-files-change

Related issue:
- https://github.com/photoprism/photoprism/issues/1252

**Reply by StephanXu** [reply]
The other issue I mentioned is that PhotoPrism can't ignore JPEG file in original directory. So even I manually proceed `photoprism convert`, it still won't generate jpeg sidecar file.

Is there an option for this?

**Reply by graciousgrey** [reply]
Currently there is no option to create additional sidecar files if one already exists in the sidecar or originals folder. 
If you want PhotoPrism to create a new sidecar file, you would need to delete the existing sidecar file or unstack the RAW from the existing JPG.

https://docs.photoprism.app/user-guide/organize/stacks/#unstack-files

What you can do is use Lightroom to create the new JPG and save it next to the RAW file [with a name that can be stacked with the existing photo](https://docs.photoprism.app/user-guide/settings/library/#stacks). Also note that PhotoPrism uses Darktable or RawTherapee to create the JPGs, which means that not all Lightroom edits may be compatible.
